### ISSUE TITLE
Cannot define a readonly global with mutable fields

### ISSUE THREAD

**Issue by hramrach** [lunarmodules/luacheck#90]
Reading the documentation I see no option to declare a global readonly (so that it cannot be replaced) but making all its fields mutable.

Understandably this is not a great interface but it's a fact of life that a runtime provides its data to lua code in this fashion.

**Comment by alerque** [comment]
I don't think it is currently possible to lint for that, nor will I likely have time or interest in figuring it out. If you can figure out a clean way to implement it I'd consider a PR contributing a new config.

**Comment by arichard4** [comment]
If there are a fixed set of fields that you'd expect to be mutable, you can explicitly mark them out as being mutable on a global read-only like so:

.luacheckrc
`stds.mutable_readonly = {
	read_globals={
		x = {
			fields = {
				example_field_name = {
					read_only = false,

				}
			}
		}
	}
}
std = "min+mutable_readonly"`

test.lua
`x = 3
x.example_field_name = 1`

> luacheck test.lua
> test.lua:1:1: setting read-only global variable x

(If the fields can be tables, you'll also want to add "other_fields = true")

This seems to only work for globals/read_globals defined in a std, not if you define them directly in the config file. Seems bad. ¯\ (ツ) /¯

**Comment by hramrach** [comment]
Yes, there is a fixed set of fields that are mutable: all possible fields.

**Comment by alerque** [comment]
That isn't what "fixed set" means. Is there a master list of all possible top level keys?

**Comment by hramrach** [comment]
It's basically access to something like the Unix environment - C strings indexed by C strings.

**Comment by ale5000-git** [comment]
There is also the opposite problem, how make the global mutable but all its field readonly?
When I extend the string library, example:
```
function string:findsomething()
end
```

and then I have some global string:
```
string1
string2
```

even objects with string fields:
`my_object.string3`

Then the string itself should be mutable but the function `findsomething` (or the functions) not.

Note: This is just an example but pose the case I have 100 functions and 100 strings, then I have to manually specify 10000 cases to make the base object read-write but fields read-only?

**Comment by ale5000-git** [comment]
Why noy something like this?

```
my_global_string = {
	read_only = false
	read_only_fields = true
	fields = {
		Something = {}
	},
}
```
